### PR TITLE
hack: add shellcheck

### PR DESCRIPTION
**Pull request by hexfusion**
I borrowed the implementation from the cluster monitor operator. With this we can add a test to CI.

### CODE REVIEW

**Comment by retroflexer** [comment]
Cool. We can manually run in our development environments but we can invoke it from CI checks as well.

/lgtm

File: hack/shellcheck.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+#TODO add make support in https://github.com/openshift/build-machinery-go
+
+if [ "${IS_CONTAINER:-}" != "" ]; then
+  TOP_DIR="${1:-.}"
+  find "${TOP_DIR}" \
+    -path "${TOP_DIR}/vendor" -prune \
+    -o -type f -name '*.sh' -exec shellcheck --format=gcc {} \+
+else
+  docker run --rm \
```
Comment (retroflexer) [inline hack/shellcheck.sh:12]: Do we want to use `podman` instead of `docker` here?

Comment (ironcladlou) [reply hack/shellcheck.sh:12]: I'd suggest making or using some existing util function to pick whichever is available for portability reasons

**Pull request merged**